### ISSUE TITLE
fixes for warnings

### ISSUE THREAD

**Issue by amaiya** [amaiya/ktrain#499]
Update `eli5` fork to address following warnings:

```shell
/home/user/mambaforge/lib/python3.9/site-packages/eli5/formatters/image.py:13: DeprecationWarning: LANCZOS is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.LANCZOS instead.
  resampling_filter=Image.LANCZOS, # type: int
/home/user/mambaforge/lib/python3.9/site-packages/eli5/formatters/image.py:242: DeprecationWarning: LANCZOS is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.LANCZOS instead.
  def expand_heatmap(heatmap, image, resampling_filter=Image.LANCZOS):
/home/user/mambaforge/lib/python3.9/site-packages/eli5/lime/textutils.py:186: DeprecationWarning: Flags not at the start of the expression '((?u)\\b\\w+\\b)' but at position 1
  parts = re.split(token_pattern, text)
/home/user/mambaforge/lib/python3.9/site-packages/sklearn/linear_model/_stochastic_gradient.py:163: FutureWarning: The loss 'log' was deprecated in v1.1 and will be removed in version 1.3. Use `loss='log_loss'` which is equivalent.
```

**Comment by amaiya** [comment]
In addition to `pandas` warnings:
```
/home/amaiya/projects/ghub/ktrain/tests/test_textregression.py:39: FutureWarning: The behavior of `series[i:j]` with an integer-dtype index is deprecated. In a future version, this will be treated as *label-based* indexing, consistent with e.g. `series[i]` lookups. To retain the old behavior, use `series.iloc[i:j]`. To get the future behavior, use `series.loc[i:j]`.
  description_train = data["description"][:train_size]
```

**Comment by amaiya** [comment]
`LANCZOZ` and `log_loss` warnings were addressed in `v037.5`.

**Comment by amaiya** [comment]
Reference to `flags not at start of expression: https://stackoverflow.com/questions/72284064/regex-expressions-deprecation-warning

**Comment by SaifBattah** [comment]
can you provide me the  'data/arabic_hotel_reviews/balanced-reviews.txt' file @amaiya

**Comment by dcisic** [comment]
import os
os.environ["TF_USE_LEGACY_KERAS"] = "1"
import warnings, gc
warnings.filterwarnings("ignore")
import tensorflow as tf

from ktrain import text


gives error:
`keras.optimizers.legacy` is not supported in Keras 3. When using `tf.keras`, to continue using a `tf.keras.optimizers.legacy` optimizer, you can install the `tf_keras` package (Keras 2) and set the environment variable `TF_USE_LEGACY_KERAS=True` to configure TensorFlow to use `tf_keras` when accessing `tf.keras`.

**Comment by amaiya** [comment]
@dcisic: If using `tensorflow>=2.16`, please install the `tf_keras` package (a version that matches your TF version) in addition to  setting the environment variable TF_USE_LEGACY_KERAS="1".  If you are using `tensorflow<2.16`, then uninstall `tf_keras` and do not set `TF_USE_LEGACY_KERAS`.  If you still have problems open a **new and separate** issue with more details including a reproducible example.
